webapp/src/client/client.ts

```diff
@@ -5,30 +5,32 @@ import {id} from '../manifest';
 
 export default class Client {
     url: string;
-    baseUrl: string = '';
+    baseUrl: Promise<any> | string = '';
 
     constructor() {
         this.url = '/plugins/' + id;
-        if ((window as any).basename) {
-            this.baseUrl = (window as any).basename;
-        }
-        this.url = this.baseUrl + '/plugins/' + id;
+        this.getBaseURL();
+    }
+
+    getBaseURL = async () => {
+        this.baseUrl = this.baseUrl === '' ? Client4.getConfig().then((config) => config.ServiceSettings.SiteURL) : this.baseUrl;
```

Comment:
There is no need to use Client4 to ask for this information to the server. We can get the information in the redux store. You can see how the Zoom plugin gets the base URL here:
https://github.com/mattermost/mattermost-plugin-zoom/blob/master/webapp/src/index.js

Reply:
We can make it so the `Client` class defined in this file is initialized with the redux store, so it always has access to the state containing the site URL